topics/wallet-standard.md

```diff
@@ -0,0 +1,248 @@
+        "appMetadata": {
+            "name": "My DApp"
+        },
+        "networks": [{
+            "chainId": "icp:737ba355e855bd4b61279056603e0550",
+        }],
+        "scopes": ["canister_call"],
+        "challenge": "" /* TODO */
+    }
+}
+
+// Response
+{
+    "id": 1,
+    "jsonrpc": "2.0",
+    "result": {
+        "version": "1",
+        "senderId": "" /* TODO */,
+        "appMetadata": {
+            "name": "My Wallet"
+        },
+        "networks": [{
+            "chainId": "icp:737ba355e855bd4b61279056603e0550"
+        }],
+        "scopes": ["canister_call"],
+        "principal": "" /* TODO */,
+        "challenge": "" /* TODO */,
+        "signature": "" /* TODO */
+    }
+}
+```
+
+### Canister Call
+
+Once the connection between the relying party and wallet is established, the relying party can request the wallet to execute canister calls.
+
+#### Types
+
+- `text`: A plain `string` value.
+- `blob`: A `string` value describing binary data encoded in base64.
+- `nat`: A `string` value of an unsigned 64-bit integer.
+- `HashTree`: A recursive type defined as a union of:
+    - `[0]`: Empty Node.
+    - `[1, HashTree, HashTree]`: Fork Node.
+    - `[2, blob, HashTree]`: Labeled Node.
+    - `[3, blob]`: Leaf Node.
+    - `[4, blob]`: Pruned Node.
+
+#### Request
+
+`version` (`text`): The version of the standard used.
+
+`senderId` (`text`): The ID of the sender. This should be used to identify the sender of the message.
+
+`network`: Network details on which the call should be executed.
+- `chainId` (`text`) - The chain ID of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`canisterId` (`text`): The ID of the canister on which the call should be executed.
+
+`method` (`text`): The name of the call method to be executed.
+
+`arg` (`blob`): The arguments for the call.
+
+
+#### Response
+
+`version` (`text`): The version of the standard used.
+
+`senderId` (`text`): The ID of the sender. This should be used to identify the sender of the message.
+
+`network`: Network details on which the call was executed.
+- `chainId` (`text`): The chain ID of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`contentMap`: The actual request content as specified [here](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-call).
+- `request_type` (`text`)
+- `sender` (`blob`)
+- `nonce` (`blob`, optional)
+- `ingress_expiry` (`nat`)
+- `canister_id` (`blob`)
+- `method_name` (`text`)
+- `arg` (`blob`)
+
+`certificate`: The certificate returned by the `read_state` call as specified [here](https://internetcomputer.org/docs/current/references/ic-interface-spec/#certificate).
```

Comment:
We should probably also explain _why_ this is the certificate and how the relying party can verify that the response is genuine and matches the request. This can probably be just a few links into existing specifications.